OOP/OOP6_ResizableArray/src/main/java/LukCharList.java

```diff
@@ -0,0 +1,163 @@
+        body = rewriteCharArray(body, new char[this.length() + 1]);
+        body[this.length() - 1] = aChar;
+        return this.length() - 1;
+    }
+
+    private char[] rewriteCharArray(char[] original, char[] copy) {
+        if (original.length > 0) {
+            for (int index = 0; index < original.length; index++) {
+                copy[index] = original[index];
+            }
+        }
+        return copy;
+    }
+
+    private char[] rewriteAndDeleteChar(char[] array, int indexOfDeleted) {
+        char[] temp = new char[this.length() - 1];
+        int arrayIndex = 0;
+
+        for (int index = 0; index < this.length() - 1; index++) {
+            if (index == indexOfDeleted) {
+                arrayIndex++;
+                temp[index] = array[arrayIndex];
+            } else {
+                temp[index] = array[arrayIndex];
+            }
+            arrayIndex++;
+        }
+
+        return temp;
+    }
+
+    public char getByIndex(int index) {
+        return body[index];
+    }
+
+    public char[] getAll() {
+        return body;
+    }
+
+    public char delete(int index) {
+        char[] temporaryArray = rewriteCharArray(body, new char[this.length()]);
+        char deletedChar = this.getByIndex(index);
+        body = rewriteAndDeleteChar(temporaryArray, index);
+        return deletedChar;
+    }
+
+    public int length() {
+        return body.length;
+    }
+
+    public int countChar(char aChar) {
+        int numberOfChar = 0;
+
+        for (int index = 0; index < this.length(); index++) {
+            if (this.getByIndex(index) == aChar) {
+                numberOfChar++;
+            }
+        }
+
+        return numberOfChar;
+    }
+
+    public int findFirst(char aChar) {
+        for (int index = 0; index < this.length(); index++) {
+            if (this.getByIndex(index) == aChar) {
+                return index;
+            }
+        }
+        return -1;
+    }
+
+    public int findLast(char aChar) {
+        for (int index = this.length() - 1; index >= 0; index--) {
+            if (this.getByIndex(index) == aChar) {
+                return index;
+            }
+        }
+        return -1;
+    }
+
+    private int[] rewriteIntArray(int[] temporaryArray, int[] finalArray) {
+        for (int index = 0; index <= finalArray.length - 1; index++) {
+            finalArray[index] = temporaryArray[index];
+        }
+        return finalArray;
+    }
+
+    public int[] findAllIndex(char aChar) {
+        int numberOfIndexes = 0;
+        int[] temporaryArray = new int[this.length()];
+        int jIndex = 0;
+
+        for (int index = 0; index < this.length(); index++) {
+            if (this.getByIndex(index) == aChar) {
+                numberOfIndexes++;
+                temporaryArray[jIndex] = index;
+                jIndex++;
+            }
+        }
+        int[] finalArray = new int[numberOfIndexes];
+        return rewriteIntArray(temporaryArray, finalArray);
```

Comment:
no need for a `finalArray` variable. Can create it in method call directly

Reply:
@MatDls modified